R/expectation.r

```diff
@@ -10,19 +10,76 @@
 #' @keywords internal
 #' @export
 expectation <- function(passed, failure_msg, srcref = NULL) {
-  structure(
+  new_expectation(passed = passed, failure_msg = failure_msg,
+                  srcref = srcref)
+}
+
+new_expectation <- function(failure_msg, srcref, ...,
+                            passed = FALSE, error = FALSE, skipped = FALSE) {
+  if (passed) {
+    class = c("expectation", "condition")
+  } else {
+    class = c("expectation", "error", "condition")
+  }
+
+  exp <- structure(
     list(
       passed = passed,
-      error = FALSE,
-      skipped = FALSE,
-      failure_msg = failure_msg,
-      srcref = srcref
+      error = error,
+      skipped = skipped,
+      failure_msg = failure_msg
     ),
-    class = "expectation"
+    class = class
   )
+
+  update_expectation(exp, srcref)
+}
+
+update_expectation <- function(exp, srcref, info = NULL, label = NULL) {
```

Comment:
It feels like we should be able to eventually eliminate this function, wrapping it into the constructor